Script manager

white-axe opened the pull request:
**Connections**
* Closes #28

**Description**
Adds a "Script Manager" under the Tools menu in the top bar that can be used to extract the scripts from Scripts.rxdata, create Scripts.rxdata from scripts or convert Scripts.rxdata to another format (JSON/YAML/RON).

**Testing**
I tested that it's capable of extracting the scripts from the following example games from https://www.rpgmakerweb.com/additional-downloads:
* KNight-Blade (RPG Maker XP)
* Legionwood (RPG Maker VX)
* Crysalis (RPG Maker VX Ace) 

**Checklist**

- [x] Run `cargo fmt`.
- [x] Run `cargo clippy`. If applicable, add:
  - [ ] `--target wasm32-unknown-unknown -Z build-std=std,panic_abort`
- [x] Run `cargo build --release`
- [ ] If applicable, run `trunk build --release`

white-axe committed:
feat: implement script manager for RPG Maker XP

white-axe committed:
feat: script manager now works for VX/VXA scripts as well

white-axe committed:
fix: fix error when extracting every script at the same time

white-axe committed:
feat: script manager now supports JSON/YAML/RON

white-axe committed:
feat: add scripts conversion to script manager

white-axe committed:
chore: clippy

melody-rs commented:
This is awesome!
It'd be really nice to be able to extract scripts from an `rgssad`, or the currently loaded project though.

The order of scripts is pretty important as well- `Main` (or an equivalent) has to be the last loaded script, so it'd be really nice if the tool could order scripts too!

white-axe commented:
The extract mode already does try loading the scripts file from the currently loaded project by default. I'll add the ability to select a .rgssad archive and have it look for the scripts file inside the archive automatically, though. As for the ordering of the scripts, I'll make it write the order of the scripts to _scripts.txt in the output folder when extracting and then read _scripts.txt from the input folder when creating.

white-axe committed:
feat: allow selecting .rgssad archive as scripts file

In the "Extract" and "Convert" parts of the script manager, you can now
select a .rgssad archive. It'll look for a scripts file inside that
archive automatically.

white-axe committed:
feat: read/write script order to _scripts.txt

melody-rs commented:
@zimberzimber since you requested this feature, I figured you might want to review it?

[luminol.zip](https://github.com/user-attachments/files/15743636/luminol.zip)

white-axe committed:
fix: fix edge case in `get_scripts_from_filesystem`

This fixes a bug where if the project's script path is set to something
beginning with "Game", the script manager searches for that in the root
directory of the project instead of the Data directory of the project.

The script manager now searches for the project's script path in the
Data directory regardless of what the script path is set to.

zimberzimber reviewed (approved):
seemsgood
